src/java.base/share/classes/jdk/internal/module/ModuleInfoWriter.java

```diff
@@ -46,122 +48,124 @@ public final class ModuleInfoWriter {
-            ModuleDescriptor.Modifier.MANDATED, ACC_MANDATED
+            ModuleDescriptor.Modifier.OPEN, Classfile.ACC_OPEN,
+            ModuleDescriptor.Modifier.SYNTHETIC, Classfile.ACC_SYNTHETIC,
+            ModuleDescriptor.Modifier.MANDATED, Classfile.ACC_MANDATED
         );
 
     private static final Map<ModuleDescriptor.Requires.Modifier, Integer>
         REQUIRES_MODS_TO_FLAGS = Map.of(
-            ModuleDescriptor.Requires.Modifier.TRANSITIVE, ACC_TRANSITIVE,
-            ModuleDescriptor.Requires.Modifier.STATIC, ACC_STATIC_PHASE,
-            ModuleDescriptor.Requires.Modifier.SYNTHETIC, ACC_SYNTHETIC,
-            ModuleDescriptor.Requires.Modifier.MANDATED, ACC_MANDATED
+            ModuleDescriptor.Requires.Modifier.TRANSITIVE, Classfile.ACC_TRANSITIVE,
+            ModuleDescriptor.Requires.Modifier.STATIC, Classfile.ACC_STATIC_PHASE,
+            ModuleDescriptor.Requires.Modifier.SYNTHETIC, Classfile.ACC_SYNTHETIC,
+            ModuleDescriptor.Requires.Modifier.MANDATED, Classfile.ACC_MANDATED
         );
 
     private static final Map<ModuleDescriptor.Exports.Modifier, Integer>
         EXPORTS_MODS_TO_FLAGS = Map.of(
-            ModuleDescriptor.Exports.Modifier.SYNTHETIC, ACC_SYNTHETIC,
-            ModuleDescriptor.Exports.Modifier.MANDATED, ACC_MANDATED
+            ModuleDescriptor.Exports.Modifier.SYNTHETIC, Classfile.ACC_SYNTHETIC,
+            ModuleDescriptor.Exports.Modifier.MANDATED, Classfile.ACC_MANDATED
         );
 
     private static final Map<ModuleDescriptor.Opens.Modifier, Integer>
         OPENS_MODS_TO_FLAGS = Map.of(
-            ModuleDescriptor.Opens.Modifier.SYNTHETIC, ACC_SYNTHETIC,
-            ModuleDescriptor.Opens.Modifier.MANDATED, ACC_MANDATED
+            ModuleDescriptor.Opens.Modifier.SYNTHETIC, Classfile.ACC_SYNTHETIC,
+            ModuleDescriptor.Opens.Modifier.MANDATED, Classfile.ACC_MANDATED
         );
 
     private static final String[] EMPTY_STRING_ARRAY = new String[0];
 
     private ModuleInfoWriter() { }
 
+    private static int modifiersToFlags(Set<ModuleDescriptor.Modifier> mods) {
+        int flags = 0;
+        if (mods.contains(ModuleDescriptor.Modifier.OPEN)) flags |= Classfile.ACC_OPEN;
+        if (mods.contains(ModuleDescriptor.Modifier.SYNTHETIC)) flags |= Classfile.ACC_SYNTHETIC;
+        if (mods.contains(ModuleDescriptor.Modifier.MANDATED)) flags |= Classfile.ACC_MANDATED;
+        return flags;
+    }
+
     /**
      * Writes the given module descriptor to a module-info.class file,
      * returning it in a byte array.
      */
     private static byte[] toModuleInfo(ModuleDescriptor md,
                                        ModuleResolution mres,
                                        ModuleTarget target) {
-        ClassWriter cw = new ClassWriter(0);
-        cw.visit(Opcodes.V10, ACC_MODULE, "module-info", null, null, null);
-
-        int moduleFlags = md.modifiers().stream()
-                .map(MODULE_MODS_TO_FLAGS::get)
```

Comment:
The `MODULE_MODS_TO_FLAGS` can be reused in `modifiersToFlags`. Otherwise, it's useless and can be removed.

Reply:
`modifiersToFlags` method is now directly inlined using `MODULE_MODS_TO_FLAGS` stream mapping
Thank you.